### PR TITLE
ASSETS-33747 named smartcrop in image core v3 component

### PR DESCRIPTION
**Pull request by mohiaror**
Instead of fastly smartcrops, expect named smartcrops in delivery URLs.

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java [hunk 1]
```diff
@@ -32,6 +43,8 @@
 import org.apache.sling.models.annotations.injectorspecific.OSGiService;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import org.json.JSONException;
```
Comment (kwin) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java:46]: You should no longer use those deprecated dependencies (https://javadoc.io/doc/com.adobe.aem/uber-jar/latest/org/json/package-summary.html), but rather `javax.json`

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java [hunk 2]
```diff
@@ -135,6 +154,37 @@ public String getSrcset() {
...
+                } else if (ngdmImage && StringUtils.equals(smartCropRendition, SMART_CROP_AUTO)) {
+                    String endPointUrl = "https://" + nextGenDynamicMediaConfig.getRepositoryId() + metadataDeliveryEndpoint;
+                    HttpGet get = new HttpGet(endPointUrl);
+                    get.setHeader("X-Adobe-Accept-Experimental", "1");
+                    try {
+                        HttpResponse response = client.execute(get);
+                        StatusLine statusLine = response.getStatusLine();
+                        int statusCode = statusLine.getStatusCode();
+                        if (statusCode == HttpStatus.SC_OK) {
+                            HttpEntity e = response.getEntity();
```
Comment (kwin) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java:166]: You should always close the response. Better use the `execute` method taking a `ResponseHandler` (https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/HttpClient.html#execute(org.apache.http.client.methods.HttpUriRequest,%20org.apache.http.client.ResponseHandler)) and directly act on the returned input stream.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java [hunk 3]
```diff
@@ -135,6 +155,37 @@ public String getSrcset() {
                 // in case of dm image and auto smartcrop the srcset needs to generated client side
                 if (dmImage && StringUtils.equals(smartCropRendition, SMART_CROP_AUTO)) {
                     srcSet = EMPTY_PIXEL;
+                } else if (ngdmImage && StringUtils.equals(smartCropRendition, SMART_CROP_AUTO) && client != null) {
+                    String endPointUrl = "https://" + nextGenDynamicMediaConfig.getRepositoryId() + metadataDeliveryEndpoint;
```
Comment (vladbailescu) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java:159]: Please move this to a separate method to facilitate unit-testing!

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java (hunk 1 above)
Comment (mohiaror) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java]: Refactored to remove the deprecated dependencies.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java (hunk 3 above)
Comment (mohiaror) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java]: Moved it into a separate function.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java (hunk 2 above)
Comment (mohiaror) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java]: @kwin Thanks. Incorporated the feedback in new commit.

**Comment by mohiaror** [comment]
Closing this PR in favor of https://github.com/adobe/aem-core-wcm-components/pull/2698

**Pull request closed without merging**